src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/representationFunctions.kt

```diff
@@ -0,0 +1,67 @@
+package io.curity.identityserver.plugin.frejaeid.authentication
+
+import se.curity.identityserver.sdk.haapi.*
+import se.curity.identityserver.sdk.http.HttpMethod
+import se.curity.identityserver.sdk.http.MediaType
+import se.curity.identityserver.sdk.web.LinkRelation
+import se.curity.identityserver.sdk.web.Representation
+import java.net.URI
+
+class GetRepresentationFunction : RepresentationFunction
+{
+    override fun apply(model: RepresentationModel, factory: RepresentationFactory): Representation
+    {
+        val authUrl = URI.create(model.getString("_authUrl"))
+        val userInfoType = model.getString("userInfoType")
+
+        return factory.newAuthenticationStep { step ->
+            step.addMessage(Message.ofKey("view.$userInfoType.description"), HaapiContract.MessageClasses.INFO)
+            step.addFormAction(HaapiContract.Actions.Kinds.LOGIN, authUrl, HttpMethod.POST,
+                    MediaType.X_WWW_FORM_URLENCODED, Message.ofKey("view.title"), Message.ofKey("view.submit")) { fields ->
```

Comment:
Nit: both these two messages can be "static", since the key is a constant (e.g use the companion object or define them as private variables outside the class.